frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -395,10 +395,20 @@ impl<T: Config> Pallet<T> {
 		// Any unprocessed solution is pointless to even consider. Feasible or malicious,
 		// they didn't end up being used. Unreserve the bonds.
 		let discarded = all_submissions.len();
-		for SignedSubmission { who, deposit, .. } in all_submissions.drain() {
+		let mut count = 0;
+		for SignedSubmission { who, deposit, call_fee, .. } in all_submissions.drain() {
+			if T::SignedMaxRefunds::get().map_or(true, |max| count < max) {
+				// Refund fee
+				let _ = T::Currency::deposit_creating(&who, call_fee);
+				count += 1;
+			}
+
+			// Unreserve deposit
 			let _remaining = T::Currency::unreserve(&who, deposit);
-			weight = weight.saturating_add(T::DbWeight::get().writes(1));
 			debug_assert!(_remaining.is_zero());
+
+			Self::finalize_signed_phase_handle_refund(&who, deposit, call_fee);
```

Comment:
you are doing a double `unreserve` here. Once in this function and once in the line above.

Reply:
`finalize_signed_phase_handle_refund` is opaque in what it does IMO, I would call it `fn refund_deposit_and_fee`, this would have made the issue clearer.